### DISCUSSION TITLE
Plugin state events

### DISCUSSION THREAD

**Discussion by borkowsky** [Annoraaq/grid-engine#68] [Ideas]
Hi. I want to suggest to add plugin state events such as 'created', 'ready' etc. I need it to know, can i use plugin's methods. Player of my game can log in game while i load scene, and addCharacter method maybe not available. Sorry for my english, its not my native language :)

**Comment by splashsky** [comment]
You should put this into an Issue so it can be tracked. These may be useful events, depending on how we can use them!

**Comment by Annoraaq** [comment]
Hi @borkowsky, sorry for the late reply. I overlooked this one.

I am not sure that I got the question right. So please correct me if I misunderstood. 
The plugin can be used as soon as you called the `create()` method on it. Since it's a synchronous method, there is no need to wait for an event that signals the state of the plugin.

You are describing the case of loading the scene. In this case the plugin is the wrong place for that functionality. But this could be interesting for you:

https://labs.phaser.io/view.html?src=src/loader\loader%20events\load%20progress.js
